docs/src/modules/components/MarkdownElement.js

```diff
@@ -380,38 +380,50 @@ const Root = styled('div')(
       fontSize: 10,
       '&:hover': {
         '& .MuiCode-copy': {
-          display: 'block',
+          display: 'inline-flex',
+          flexDirection: 'row-reverse',
         },
       },
     },
     '& .MuiCode-copy': {
-      minWidth: 64,
       display: 'none',
-      backgroundColor: alpha(lightTheme.palette.primaryDark[600], 0.5),
+      width: 30,
       cursor: 'pointer',
       position: 'absolute',
       top: theme.spacing(1),
       right: theme.spacing(1),
       fontFamily: 'inherit',
-      fontSize: lightTheme.typography.pxToRem(13),
       fontWeight: 500,
-      padding: theme.spacing(0.5, 1),
-      borderRadius: 4,
+      padding: theme.spacing(0.5),
+      borderRadius: 6,
       border: `1px solid`,
       borderColor: lightTheme.palette.primaryDark[500],
+      backgroundColor: lightTheme.palette.primaryDark[600],
       color: lightTheme.palette.primaryDark[50],
+      '& svg': {
+        userSelect: 'none',
+        width: '1em',
+        height: '1em',
+        display: 'inline-block',
+        fill: 'currentcolor',
+        flexShrink: 0,
+        fontSize: '18px',
+      },
       '&:hover, &:focus': {
         opacity: 1,
         color: '#fff',
-        backgroundColor: alpha(lightTheme.palette.primaryDark[600], 0.7),
-        borderColor: lightTheme.palette.primaryDark[500],
+        backgroundColor: lightTheme.palette.primaryDark[500],
+        borderColor: lightTheme.palette.primaryDark[400],
         '& .MuiCode-copyKeypress': {
           display: 'block',
           // Approximate no hover capabilities with no keyboard
           // https://github.com/w3c/csswg-drafts/issues/3871
           '@media (any-hover: none)': {
             display: 'none',
           },
+          '& .MuiCode-copied-icon': {
+            display: 'none',
+          },
         },
       },
       '& .MuiCode-copyKeypress': {
```

Comment:
```suggestion
        '& .MuiCode-copyKeypress': {
          display: 'block',
          // Approximate no hover capabilities with no keyboard
          // https://github.com/w3c/csswg-drafts/issues/3871
          '@media (any-hover: none)': {
            display: 'none',
          },
        },
      },
      '& .MuiCode-copied-icon': {
        display: 'none',
      },
      '& .MuiCode-copyKeypress': {
```

The SVG element is not inside the `.MuiCode-copyKeypress` so the rule you set never impacted the element
![image](https://github.com/mui/material-ui/assets/45398769/7f2875aa-38c0-4c85-ad62-6fe145d28ae9)


Out of the topic, it could be nice to merge the two `'& .MuiCode-copyKeypress': {` definitions.